### PR TITLE
Add more unit tests for filter and service recipes, update naming to match styles

### PR DESCRIPTION
**Pull request by drenalin23**
# What
Added the rest of the a set of basic unit tests for the filter and service recipes.  There should now be 100% coverage for all of the basic resources in the current cookbook/recipes.
Updated naming conventions for tests and recipes where they did not match prevailing convention for the rest of the files.

# How
Added chefspec tests in the test/unit/spec folder.  Updated the name of all tests to match prevalent naming style.  
With ChefDK installed run $ chef exec rspec test/unit 
to run all of the unit tests.

# Why 
To finish setting up basic unit testing for the repository.  To continue to cleanup any deviations in naming or style so that the recipes remain cleanly named.

# TODO
I should either remove the renaming of the connection-pool recipe OR should finish renaming the attribute names in the default files.  It could also be pulled into a separate PR.  Please comment on this part of the PR in particular and I'll adjust it.

### CODE REVIEW

**Comment by filler** [comment]
:+1: lgtm! :rocket:

Would you like to be added as a curator of this book so you can manage PRs? It looks like MMI is doubling down which is awesome.

**Comment by drenalin23** [comment]
Nick sounds good if you want to add me as someone to help review/approve/merge PRs.  We are using this cookbook for both Metrics and Monitoring so we'll continue to enhance and improve as time goes on.

**Comment by drenalin23** [comment]
Feel free to add mburns and jguice as well from the Monitoring side.  We wrote up a wrapper cookbook (mmi-cookbooks/ele-repose) which duplicates some of what should be pushed back upstream to this cookbook.

**Comment by filler** [comment]
:cool: :sunglasses: 

I poked the Repose Ninjas this AM to get the following Rackers added to the team which allows push/merge of `rackerlabs/cookbook-repose`:

@mburns 
@drenalin23 
@jguice 
@usnavi 

:beers:

File: CHANGELOG.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (filler) [inline CHANGELOG.txt]: Awwwwww yisssss!

**Comment by filler** [comment]
@drenalin23 do you have authz to click 'Merge' now?  Repose Ninjas should have provisioned authz for the 4 x MMI Rackers above.

**Comment by filler** [comment]
Yaaaaaay! Do great things fiends!

**Pull request merged**